python/test/unit/language/test_core.py

```diff
@@ -1151,7 +1153,7 @@ def test_index1d(expr, dtype_str, num_ctas, device):
     rank_y = expr.count(',') + 1
     shape_x = [32 for _ in range(rank_x)]
     shape_z = [32 for _ in range(rank_y)]
-    shape_z_rank_mismatch = [32 for _ in range(rank_y + 1)]
+    shape_z_rank_mismatch = [32 for _ in range(rank_y - 1)]
```

Comment:
It turns out `broadcast_impl_shape` actually doesn't handle rank mismatches, so previously store didn't support e.g. storing a 2d value to 3d pointer. This change now makes it legal, so I change the test to check that storing a 3d value to a 2d pointer raises an error. This I explicitly ban because it means you're storing multiple values to the same address and is always a race.